packages/wallet/src/services/DelegationTracker/RewardsHistory.ts

```diff
@@ -34,11 +34,20 @@ export const createRewardsHistoryProvider =
 
 export type RewardsHistoryProvider = ReturnType<typeof createRewardsHistoryProvider>;
 
-const firstDelegationEpoch$ = (transactions$: Observable<TxWithEpoch[]>) =>
+const firstDelegationEpoch$ = (transactions$: Observable<TxWithEpoch[]>, rewardAccounts: Cardano.RewardAccount[]) =>
   transactions$.pipe(
     map((transactions) =>
       first(
-        transactions.filter(({ tx }) => transactionHasAnyCertificate(tx, [Cardano.CertificateType.StakeDelegation]))
+        transactions.filter(({ tx }) => {
+          const hasCertificate = transactionHasAnyCertificate(tx, [Cardano.CertificateType.StakeDelegation]);
```

Comment:
I think this is redundant, as `sentInspector` (almost) does the same job. We might be able to delete `transactionHasAnyCertificate` because it's not that useful without checking rewardAccounts. It should probably be replaced with an inspector. Also, deleting `transactionHasAnyCertificate` will probably flesh out some other similar issues to fix.

Reply:
I added the inspector: 
```ts
SignedCertificatesInspector = (
  rewardAccounts: RewardAccount[],
  certificateTypes?: CertificateType[]
) => Inspector<Certificate[]>;
```

but `transactionHasAnyCertificate` is used by `transactionsWithCertificates` and that one by `DelegationTracker.certificateTransactionsWithEpochs`.

I could delete `transactionHasAnyCertificate` and change `transactionsWithCertificates` to receive reward accounts and use the inspector but that would only return the transactions with a certificate signed by those accounts, is that ok?

Reply:
Yes, I think in wallet context we always only care about own certificates